### ISSUE TITLE
Excessive Log Entries in OTel Collector with OTLP-HTTP Traffic

### ISSUE THREAD

**Issue by bvsvas** [open-telemetry/opentelemetry-collector#11336]
**Describe the bug**
We are experiencing a high volume of redundant log entries in the OTel Collector when handling OTLP-HTTP traffic. This log noise makes it difficult to visualize and focus on key error messages. These log entries appear frequently when the OTLP receiver is processing HTTP traffic.
**Example log output:**
```
2024/10/02 12:40:20 http: superfluous response.WriteHeader call from go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal/request.(*RespWriterWrapper).writeHeader (resp_writer_wrapper.go:78)
2024/10/02 12:40:20 http: superfluous response.WriteHeader call from go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal/request.(*RespWriterWrapper).writeHeader (resp_writer_wrapper.go:78)
2024/10/02 12:40:20 http: superfluous response.WriteHeader call from go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal/request.(*RespWriterWrapper).writeHeader (resp_writer_wrapper.go:78)
```

**Steps to reproduce**
Send HTTP traffic with OTLP receiver. 

OTel Collector
1. OTLP receiver
2. Debug Exporter

**What did you expect to see?**
This noise hinders the ability to quickly identify and address significant errors, leading to potential oversight of critical log entries during debugging.

**What did you see instead?**
Too many warning entires flooding the log. 

**What version did you use?**
OTel Collector: v0.109.0

**What config did you use?**

```yaml
extensions:
  health_check:
    endpoint: ${env:OTEL_POD_IP}:13133

exporters:
  debug: {}

processors:
  memory_limiter:
    limit_mib: 6500
  probabilistic_sampler:
    sampling_percentage: 30

receivers:
  otlp:
    protocols:
      grpc:
        endpoint: ${env:OTEL_POD_IP}:4317
      http:
        endpoint: ${env:OTEL_POD_IP}:4318

service:
  pipelines:
    metrics:
      exporters:
        - debug
      processors:
        - memory_limiter
        - cumulativetodelta
      receivers:
        - otlp
        - prometheus
    traces:
      exporters:
        - apmexporter
      processors:
        - memory_limiter
      receivers:
        - otlp

```

**Environment**
OS: CentOS v8
GoLang: 1.23

**Additional context**

**Comment by bvsvas** [comment]
receivers/otlp

**Comment by olabodeIdowu** [comment]
Hi, @bvsvas Hoping this solution will help you

There are times when you want to reduce log volume being sent to your centralized logging backend. You may want to do that for a variety of reasons:
1.0 To reduce the cost associated with your observability backend as some of them can become very expensive with large log volumes.
2.0 To reduce the noise in your observability backend which is your reason
3.0 To reduce the amount of data being sent to your observability backend to reduce the load on your network.

Processors are a way to transform incoming data. They can be used to filter, transform, enrich, sample, etc. incoming data. They can be used to transform logs, metrics and traces. You can find the list of all processors here. The 2 processors of interest to us for filtering logs are:

1.0 Filter Processor - Allows to filter the complete log records based on the conditions specified.
2.0 Attributes Processor - Allows to filter attributes from the log records. e.g. log_file_path

Lets talk about this first example of processor Filter Processor since the question are about log entries

**An example filter logs at source** 

Let's say you have a log record that looks like this:

```
{
  "_timestamp": 1702059473796197,
  "body": "time=\"2023-12-08T18:17:53Z\" level=info
    msg=\"finished unary call with code OK\"
    grpc.code=OK
    grpc.method=Check
    grpc.service=grpc.health.v1.Health
    grpc.start_time=\"2023-12-08T18:17:53Z\"
    grpc.time_ms=0.02 span.kind=server system=grpc",
  "dropped_attributes_count": 0,
  "k8s_container_name": "argocd-repo-server",
  "k8s_container_restart_count": "0",
  "k8s_deployment_name": "argocd-repo-server",
  "k8s_namespace_name": "argocd",
  "k8s_node_name": "ip-10-2-147-27.us-east-2.compute.internal",
  "k8s_pod_name": "argocd-74ff5598cd-zjsxz",
  "k8s_pod_start_time": "2023-11-02 21:59:53 +0000 UTC",
  "k8s_pod_uid": "6b3f057f-c4ac-4d65-8c03-262979815daa",
  "log_file_path": "/var/log/containers/argocd-74ff5598cd-zjsxz_argocd-262979815daa.log",
  "log_iostream": "stderr",
  "logtag": "F",
  "service_name": "argocd-repo-server",
  "severity": "",
  "time": "2023-12-08T18:17:53.79619795Z"
}

```
**Filter log records for a specific namespace (Using filter processor)**

You would like to filter out all logs that are from the `**argocd**` namespace. You can do that by:

using **OTTL conditions**:

```
processors:
  filter/remove_argocd_logs:
    error_mode: ignore
    logs:
      log_record:
        - resource.attributes["k8s.namespace.name"]=="argocd"
```
Remember that we replaces attribute names with . to _ and hence we used k8s.namespace.name and not k8s_namespace_name

OTTL - [OpenTelemetry Transformation Language](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/README.md) allows you to transform incoming logs, metrics and traces.

or **using resource_attributes**:

```
processors:
  filter/remove_argocd_logs:
    error_mode: ignore
    logs:
      exclude:
        match_type: strict
        resource_attributes:
          - key: k8s.namespace.name
            value: argocd
```
Once you have any one of the above filters in place, add them to your pipeline (If you miss this step and nothing will happen)

```
service:
  extensions: [zpages]
  pipelines:
    logs:
      receivers: [filelog/std]
      processors: [batch, k8sattributes, filter/remove_argocd_logs]
      exporters: [otlphttp]
```

and then redeploy your otel-collector. You should see that all logs from the **`argocd`** namespace are no longer being sent to your observability backend.

Let's dig a bit deeper into what all is possible with OTTL filters.

[open-telemetry/opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor)

**Comment by TylerHelmuth** [comment]
This issue was solved in `v0.110.0`
